wherehows-web/app/routes/application.js

```diff
@@ -150,6 +163,36 @@ export default Route.extend(ApplicationRouteMixin, {
     run.scheduleOnce('afterRender', this, 'processLegacyDomOperations');
   },
 
+  /**
+   * We are NOT using the renderTemplate hook to implement any custom rendering logic. Since we don't have a didTransition
+   * hook for the application route, this hook is being used instead to schedule a banner after initial render in order to
+   * let users see the animation for banner entrance
+   */
+  renderTemplate() {
```

Comment:
I think this approach seems more complicated and coupling unrelated concerns.
I would:
- render the component in the application template (i think you already do this)
- pass a reference to the banner service to the banner component
- toggle the rendering of banners based on the values of computedProperties in the service

having the application route responsible for initializing and rendering banners with `initializeBanners` seems like unnecessary coupling the application route.

Reply:
Ah. The original point was that it seemed like there was no good way to time the entrance of the banners to be after the page loads, but I'll see how to better separate concerns with that in mind.

Reply:
Looks like we need the application route to still handle some of the initial logic and the logic that runs the scheduled function. Otherwise, we cannot schedule the init banners to run after all the rendering ends and that causes jerky starting animations since we're using css and class toggles. But all the logic to actually create the banners is moved to the service and the application route only schedules the trigger now as a middle ground.